docs/restricting-api-access.mdx

```diff
@@ -0,0 +1,157 @@
+
+## Allowed API origins
+
+If the list of the origins you expect to call your API is known in advance, you can automatically reject the unknown ones.
+
+In the project view of your Saleor Cloud dashboard, you can customize the allowed origins. The default option is to "Allow all" but you can also specify a list of origins or set it to "Dashboard only".
+
+:::info
+
+Changing the settings to only allow the selected origins can affect the installed apps. If you want your app to call the Saleor API from the client, you need to add the app's origin to the list of allowed origins. You can read more about the app's access scopes in the [App permissions](developer/extending/apps/architecture/app-permissions.mdx#access-scopes) document.
+
+:::
+
+You don't need to specify the Saleor Cloud dashboard origin, as it is always allowed to call the Saleor API.
+
+## API Password Protection
+
+Another way to restrict access to your Saleor Cloud instance is to protect it with a password through [Basic Auth](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#basic_authentication_scheme). This way, you can ensure that only those with the credentials can access the API.
+
+:::important
+
+After enabling password protection, users trying to open the Saleor Dashboard will be prompted to enter the Basic Auth username and password besides their own Saleor Cloud credentials.
+
+:::
+
+Combined with a server-side proxy, this can be a useful tool to obscure the Saleor API from unauthorized access. We will explore this in more detail in the example below.
+
+## Example: Password-protected API proxy
+
+### Overview
+
+By default, Saleor exposes some parts of its API publicly, like products catalog or checkout. This behavior doesn't have to align with the requirements of your business. However, Saleor Cloud allows you to restrict access to the API by setting up a Basic Auth.
+
+Imagine you don't want your products to be publicly accessible. To achieve this, we will create a proxy server that will know the credentials and the API URL and thus be able to access the Saleor API. The only way to get to the Saleor API will be through that proxy server.
+
+Here is how it may look in practice in the example of a simple Next.js application. At the moment, it calls the Saleor API directly:
+
+### Implementation
+
+#### Step 1: Protect the Saleor API with a password
+
+Set up "API Password Protection" in the Saleor Cloud dashboard as [described above](#api-password-protection).
+
+#### Step 2: Visit your current implementation
+
+In the current implementation, your Next.js application calls the Saleor API directly:
+
+:::note
+
+In a production-grade Next.js application, you most likely would use some GraphQL client (like [Apollo Client](https://www.apollographql.com/docs/react/)) to fetch the data from the Saleor API. However, for simplicity, we will use the native `fetch` API in this example.
```

Comment:
One of our core value propositions is being "technology agnostic", providing Next.js as an example feels quite opinionated, I would recommend providing a more generic example

Reply:
I see your point. Because I already rewrote the example once (based on Krzyh's feedback), I only added a little Next.js info box that explains the choice. If that creates confusion in the future, I will rewrite it.

Reply:
On the other hand, NextJS might be the most common tool for creating storefronts with additional BBF functionalities. This example and docs will for sure find a broad audience and should be easy to transfer to other stacks.
I think it's good as is and we can come back to it in the future